### PR TITLE
Simpler, case-sensitive, resolver

### PR DESCRIPTION
**Pull request by alcuadrado**
This PR simplifies the `Resolver` class, while making it case-sensitive.

Its tests haven't been updated yet.

### CODE REVIEW

**Comment by alcuadrado** [comment]
This is ready for review now, @fvictorio.

There are some failing tests, but were already failing, and are unrelated to this PR.

File: packages/buidler-core/src/internal/solidity/source-names.ts [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+const NODE_MODULES = "node_modules";
+
+/**
+ * This function validates the source name's format.
+ *
+ * It throws if the format is invalid.
+ * If it doesn't throw all you know is that the format is valid.
+ */
+export function validateSourceNameFormat(sourceName: string) {
+  if (path.isAbsolute(sourceName) || sourceName.startsWith("/")) {
```
[the remark this answers is not in the document]
Comment (alcuadrado) [reply packages/buidler-core/src/internal/solidity/source-names.ts:15]: Yes, in windows. I'll add a comment. There was one, but got lost during refactoring.

File: packages/buidler-core/test/internal/solidity/resolver.ts [hunk 2]
```diff
@@ -96,500 +98,370 @@ describe("Resolved file", function () {
...
-
-      assertResolvedFile(resolved2, {
-        globalName: "contracts/subdir/C.sol",
-        absolutePath: absolutePath2,
-        content: buildContent("C"),
-        lastModificationDate: ctime2,
-        library: undefined,
-      });
-    });
+async function assertResolvedFile2(
```
[the remark this answers is not in the document]
Comment (alcuadrado) [reply packages/buidler-core/test/internal/solidity/resolver.ts:101]: Ouch, I forgot to rename it 😅 Thanks for pointing at it. Any suggestion?

File: packages/buidler-core/src/internal/core/errors-list.ts [hunk 3]
```diff
@@ -606,6 +612,76 @@ This is disabled for security reasons.`,
...
       shouldBeReported: false,
     },
+    INVALID_IMPORT_BACKSLASH: {
+      number: 408,
+      message:
+        "Invalid import %imported% from %from%. Imports must use / instead of \\",
+      title: "Invalid import: use / instead of \\",
+      description: `A Solidity file is trying to import another one with its relative path and is using backslashes (\\) insteado of slashes (/).
+      
+You must always use slashes (/) in Solidity imports.`,
```
[the remark this answers is not in the document]
Comment (alcuadrado) [reply packages/buidler-core/src/internal/core/errors-list.ts:622]: I like it

File: packages/buidler-core/test/internal/solidity/resolver.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (fvictorio) [reply packages/buidler-core/test/internal/solidity/resolver.ts:101]: Maybe something like `assertResolvedFileContent` or `assertResolvedFileData`?

Or maybe this one should be `assertResolvedFile` and the previous one should be `assertResolvedFilesEqual`?

File: packages/buidler-core/src/internal/solidity/source-names.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (alcuadrado) [reply packages/buidler-core/src/internal/solidity/source-names.ts:15]: Oh, this should have been an `isAbsoluteSourceName` call. That's where the comment it is.

File: packages/buidler-core/test/internal/solidity/resolver.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (alcuadrado) [reply packages/buidler-core/test/internal/solidity/resolver.ts:101]: I renamed both asserts

**Pull request merged**